<?xml version="1.0"?>
<launch>
  <arg name="debug" default="True"/>
  <arg name="use_ros_control" default="True"/>
  <arg name="rt_frequency" default="100.0"/>
  <arg name="ctrl_loop_frequency" default="20"/>
  <arg name="prefix" default="" />

  <!-- Load the URDF/Xacro model of our robot -->
  <include file="$(find qr_description)/launch/upload_robot.launch" />
  <!-- echo $LD_LIBRARY_PATH | awk -F':' '{split($0,arr,":")}END{for(i=1;i<=NF;i++)printf("%s\n",arr[i])}'|grep 'qr_ws' -->
  <!-- Load hardware interface -->
  <node name="qr_driver_alpha" pkg="qr_driver" type="qr_driver" output="screen">
    <!-- param name="configure" command="cat '$(find qr_driver)/config/robot_config.xml'"/ -->
    <param name="configure" command="cat '$(find qr_driver)/config/robot_config.xml'"/>
    <param name="library"   value="$(env HOME)/Workspaces/qr_ws/devel/lib/libqr_driver_sys_platform.so"/>
    <param name="use_ros_control"     type="bool"   value="$(arg use_ros_control)"     />
    <param name="debug"               type="bool"   value="$(arg debug)"               />
    <param name="rt_frequency"        type="double" value="$(arg rt_frequency)"        />
    <param name="ctrl_loop_frequency" type="double" value="$(arg ctrl_loop_frequency)" />
    <param name="prefix"                            value="$(arg prefix)"              />
  </node>

  <!-- load other controller --> 
  <!--node name="ros_control_controller_manager" pkg="controller_manager" type="controller_manager" respawn="false"
    output="screen" args="load pos_based_pos_traj_controller" /-->

  <!-- Convert joint states to /tf tranforms -->
  <!--node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/-->

</launch>

